### PR TITLE
release-20.1: security: Disable TLS 1.3

### LINKED ISSUE #48294: security: Option to configure TLS version and ciphers
CockroachDB 20.1 is the first version to use TLS 1.3 by default. This turns out to be a compatibility problem with some versions of Java, including the current LTS release (https://github.com/cockroachdb/docs/issues/7258), and our compatibility testing of TLS 1.3 isn't great (https://github.com/cockroachdb/cockroach/issues/48293) so there may be other headaches lurking.

The java incompatibility is a bug on the java side; patch releases are available but weren't released until April 2020 so they're not ubiquitous yet. There is also a workaround by setting the environment variable `GODEBUG=tls13=0`; support for this env var has been removed in Go 1.14 (and therefore it will not be supported in CRDB 20.2).

Given the compatibility impact with a major platform, should we consider giving users the option to disable TLS 1.3 (in a form other than an env var, which might be easier to manage in some environments}? Should we consider disabling it by default in 20.1 since it is evidently still going through interoperability headaches in April 2020? Note that if we make any changes to CRDB here, they would presumably appear in the patch release 20.1.1.

### PR DESCRIPTION
**Pull request by bdarnell**
The go and java implementations of TLS 1.3 do not interoperate (until
Java version 11.0.7, which was released only a couple of weeks before
CRDB 20.1). Disable TLS 1.3 in version 20.1 to avoid pain during the
transition period; it should be enabled in 20.2 once the java patch
has had time to be deployed.

Fixes #48294

Release note (bug fix): TLS 1.3 is now disabled by default to improve
compatibility with Java.

Git notes: The commit in this PR has v20.1.0-rc.2 as its ancestor, so it can be used directly for the next build. It is being merged into release-20.1 so it will be included in future patch releases. It is deliberately not going into master at this time so we can continue to test compatibility with TLS 1.3 in preparation for hopefully enabling it by default in 20.2.

### CODE REVIEW

**Comment by bdarnell** [comment]
cc @knz

**Review by petermattis — APPROVED** [review]
:lgtm:

Have we manually verified that this interoperates with the incompatible version of the JDK?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48365#-:-M6WHw0s8WrCs8VcBmOf:bfd63fw)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @aaron-crl and @petermattis)

**Comment by bdarnell** [comment]
> Have we manually verified that this interoperates with the incompatible version of the JDK?

I have not, and I'm not sure how. @timveil ?

File: build/builder.sh [hunk 1]
```diff
@@ -120,10 +120,15 @@ vols="${vols} --volume=${HOME}/.yarn-cache:${container_home}/.yarn-cache${cached
 
 # If we're running in an environment that's using git alternates, like TeamCity,
 # we must mount the path to the real git objects for git to work in the container.
-alternates_file=${cockroach_toplevel}/.git/objects/info/alternates
-if test -e "${alternates_file}"; then
-  alternates_path=$(cat "${alternates_file}")
-  vols="${vols} --volume=${alternates_path}:${alternates_path}${cached_volume_mode}"
+# alternates_file=${cockroach_toplevel}/.git/objects/info/alternates
```
Comment (knz) [inline build/builder.sh:123]: Wait why are you including this change in this PR? It seems unrelated.

Comment (bdarnell) [reply build/builder.sh:123]: It's necessary due to something that changed in teamcity. We don't want to pull in any more than we have to on top of v20.1.0-rc.2, which means we need to duplicate that commit on this PR branch.

**Comment by knz** [comment]
oh nvm I see it now. carry on

**Comment by dt** [comment]
cc @dt
